packages/app-builder-lib/src/electron/electronMac.ts

```diff
@@ -222,39 +213,44 @@ export async function createMacApp(packager: MacPackager, appOutDir: string, asa
     )
 
     // `CFBundleDocumentTypes` may be defined in `mac.extendInfo`, so we need to merge it in that case
-    appPlist.CFBundleDocumentTypes = [...(appPlist.CFBundleDocumentTypes || []), ...documentTypes]
+    appPlist.CFBundleDocumentTypes = [...((appPlist.CFBundleDocumentTypes as PlistValue[]) || []), ...documentTypes]
   }
 
   if (asarIntegrity != null) {
-    appPlist.ElectronAsarIntegrity = asarIntegrity
+    appPlist.ElectronAsarIntegrity = JSON.parse(JSON.stringify(asarIntegrity))
```

Comment:
Whats the purpose of the `stringify` -> `parse` here? This seems like it's unnecessary or might be hiding a different issue.

Reply:
How about this modification?

```javascript
  const toPlistObject = (asarIntegrity: AsarIntegrity): PlistObject => {
    const result: PlistObject = {}
    for (const [filePath, headerHash] of Object.entries(asarIntegrity)) {
      result[filePath] = {
        algorithm: headerHash.algorithm,
        hash: headerHash.hash,
      }
    }
    return result
  }

  if (asarIntegrity != null) {
    appPlist.ElectronAsarIntegrity = toPlistObject(asarIntegrity)
  }
```

Reply:
Looks great!